src/main/java/com/shapesecurity/salvation/Parser.java

```diff
@@ -291,7 +295,18 @@ private boolean eat(@Nonnull Class<? extends Token> c) {
                     }
                     result = new ReferrerDirective(referrerTokens);
                     break;
+                case ReportTo:
+                    Set<ReportToValue> reportToTokens = this.parseReportToTokenList();
+                    if (reportToTokens.isEmpty()) {
+                        this.error(token, "The report-to directive must contain exactly one token");
+                        throw INVALID_REPORT_TO_TOKEN_LIST;
+                    }
+                    result = new ReportToDirective(reportToTokens);
+                    break;
                 case ReportUri:
+                    // bump to .warn once CSP3 becomes RC
+                    this.info(token,
+                        "CSP version 3 currently plans to deprecate report-uri in favour of new report-to directive.");
```

Comment:
I know I gave you this exact text, but I would prefer "A draft of the next version of CSP deprecates report-uri in favour of a new report-to directive.". LGTM otherwise :+1:.